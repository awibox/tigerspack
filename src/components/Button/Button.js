import React from 'react';
import PropTypes from 'prop-types';
import { colors } from '../../utils/colors';
import { defaultStyles } from '../../utils/defaultStyles';

const Button = (props) => {
  const {
    children,
    icon,
    indent,
    theme,
    type,
    ...otherProps
  } = props;
  const palette = colors[theme] ? colors[theme] : colors.primary;
  const size = props.size > 0 && props.size < 6 ? props.size : defaultStyles.buttonSize;
  // Dynamic styles
  const dynamicStyles = {};
  switch (type) {
    case 'text': {
      dynamicStyles.background = 'transparent';
      dynamicStyles.backgroundHover = palette.color;
      dynamicStyles.border = '1px solid transparent';
      dynamicStyles.color = palette.color;
      dynamicStyles.colorHover = palette.hoverText;
      break;
    }
    case 'outline': {
      dynamicStyles.background = 'transparent';
      dynamicStyles.border = `1px solid ${palette.color}`;
      dynamicStyles.backgroundHover = palette.color;
      dynamicStyles.color = palette.color;
      dynamicStyles.colorHover = palette.hoverText;
      break;
    }
    default: {
      dynamicStyles.background = palette.color;
      dynamicStyles.border = '1px solid transparent';
      dynamicStyles.backgroundHover = palette.hoverColor;
      dynamicStyles.color = palette.text;
      dynamicStyles.colorHover = palette.text;
      break;
    }
  }
  // Styles
  const styles = {
    button: {
      boxSizing: 'border-box',
      position: 'relative',
      outline: 'none',
      transition: defaultStyles.animation,
      display: 'flex',
      flexDirection: 'column',
      flexFlow: 'row',
      alignContent: 'center',
      alignItems: 'center',
      justifyContent: 'center',
      justifyItems: 'center',
      marginBottom: `${indent}px`,
      fontWeight: 500,
      backgroundImage: 'none',
      whiteSpace: 'nowrap',
      userSelect: 'none',
      cursor: 'pointer',
      lineHeight: 0,
      borderRadius: type !== 'rounded' ? defaultStyles.borderRadius : '20px',
      backgroundColor: dynamicStyles.background,
      border: dynamicStyles.border,
      color: dynamicStyles.color,
      fontSize: `${size * 2 + 9}px`,
      height: `${(size * 2 + 9) + (size * 5)}px`,
      padding: `${((size * 5) / 2) - 0.5}px ${size * 5}px ${((size * 5) / 2) + 0.5}px`,
      ':disabled': {
        opacity: 0.7,
        cursor: 'not-allowed',
        ':hover': {
          color: dynamicStyles.color,
          backgroundColor: dynamicStyles.background,
        },
      },
      ':hover': {
        color: dynamicStyles.colorHover,
        textDecoration: 'none',
        backgroundColor: dynamicStyles.backgroundHover,
      },
    },
    icon: {
      marginLeft: `-${(size + 5) / 2}px`,
      marginRight: children ? `${(size + 5)}px` : `-${(size + 5) / 2}px`,
    },
  };
  return (
    <button css={styles.button} {...otherProps}>
      {icon ? <span css={styles.icon}>{icon}</span> : ''}
      {children}
    </button>
  );
};

Button.propTypes = {
  theme: PropTypes.string,
  children: PropTypes.any,
  size: PropTypes.oneOf([1, 2, 3, 4, 5]),
  type: PropTypes.oneOf(['text', 'outline', 'rounded']),
  icon: PropTypes.any,
  indent: PropTypes.number,
};

Button.defaultProps = {
  indent: 0,
};

export default Button;
