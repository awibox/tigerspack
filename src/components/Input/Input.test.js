import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

const clickFn = jest.fn();

describe('Button', () => {
  it('Button default', () => {
    const component = shallow(<Button onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
  it('check Button click', () => {
    const component = shallow(<Button onClick={clickFn}>Test button</Button>);
    component.simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
  it('small text button', () => {
    const component = shallow(<Button size={'small'} type={'text'} onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
  it('large outline button', () => {
    const component = shallow(<Button size={'large'} type={'outline'} onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
  it('rounded button with icon', () => {
    const component = shallow(<Button theme={'danger'} icon={<span>icon</span>} type={'rounded'} onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
  it('button with icon only', () => {
    const component = shallow(<Button theme={'info'} icon={<span>icon</span>} onClick={clickFn}/>);
    expect(component).toMatchSnapshot();
  });
  it('disabled button', () => {
    const component = shallow(<Button disabled onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
});
