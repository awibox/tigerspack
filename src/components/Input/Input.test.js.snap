// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input default 1`] = `
.input {
  -moz-box-sizing: border-box !important;
  box-sizing: border-box !important;
  background: #fff !important;
  border-radius: 7px !important;
  border: 2px solid #cfd1d7 !important;
  position: relative !important;
  margin-bottom: 25px !important;
}

.label {
  position: absolute !important;
  background: #fff !important;
  font-size: 12px !important;
  top: -8px !important;
  left: 15px !important;
  color: #cfd1d7 !important;
  font-weight: 600 !important;
  padding: 0 3px !important;
  display: none !important;
  -webkit-transition: display .4s, color .4s ease !important;
  -moz-transition: display .4s, color .4s ease !important;
  transition: display .4s, color .4s ease !important;
}

.control {
  background: none !important;
  border: none !important;
  outline: none !important;
  font-size: 16px !important;
  width: 100% !important;
  padding: 17px 14px !important;
  -moz-box-sizing: border-box !important;
  box-sizing: border-box !important;
}

<div
  className="input"
>
  <div
    className="label"
  >
    Test
  </div>
  <input
    autoComplete="off"
    className="control"
    placeholder="Test"
  />
</div>
`;
